components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java

```diff
@@ -410,12 +410,19 @@ public static API importApi(String extractedFolderPath, APIDTO importedApiDTO, B
                 if (deploymentNameElement != null) {
                     String deploymentName = deploymentNameElement.getAsString();
                     if (gatewayEnvironmentsSet.contains(deploymentName)) {
+                        JsonElement deploymentVhostElement = deploymentJson.get(ImportExportConstants.DEPLOYMENT_VHOST);
+                        if (deploymentVhostElement == null) {
+                            throw new APIManagementException(
+                                    String.format("Required field \"%s\" is not defined in the deployment",
```

Comment:
instead of throwing an exception, can't we assign the default value localhost?

Reply:
Yes, Vhost should be an optional value.

Reply:
+1

Reply:
Thanks all fixed with https://github.com/wso2/carbon-apimgt/pull/10140/commits/2534b78073e669303fd191cb5f72e3fcbc59faee